app/main.py

```diff
@@ -1,29 +1,61 @@
+import json
 import logging
 import sys
 
+from fastapi.exceptions import RequestValidationError
 from fastapi.logger import logger
 from fastapi.openapi.utils import get_openapi
+from fastapi.requests import Request
+from fastapi.responses import JSONResponse
 from fastapi.staticfiles import StaticFiles
 from starlette.middleware.base import BaseHTTPMiddleware
 
 from app.routes.tasks import tasks
 
 from .application import app
+from .errors import ClientError, ServerError
 from .middleware import redirect_latest, set_db_mode
 from .routes import security
 from .routes.features import features
 from .routes.geostore import geostore
 from .routes.meta import assets, datasets, versions
 from .routes.sql import queries
 
-###############
+################
 # LOGGING
 ################
 
 gunicorn_logger = logging.getLogger("gunicorn.error")
 logger.handlers = gunicorn_logger.handlers
 sys.path.extend(["./"])
 
+
+################
+# ERRORS
+################
+
+
+@app.exception_handler(ClientError)
+async def client_error_handler(request: Request, exc: ClientError):
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "fail", "data": exc.detail}
+    )
+
+
+@app.exception_handler(ServerError)
+async def server_error_handler(request: Request, exc: ServerError):
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "error", "message": exc.detail}
+    )
```

Comment:
What happens if you make handle the HTTP Exception directly?

Reply:
Doing so now. Seems to work. Will remove the other handlers + errors in the future as discussed.